mlir/lib/Transforms/RemoveDeadValues.cpp

```diff
@@ -433,6 +433,19 @@ static void processRegionBranchOp(RegionBranchOpInterface regionBranchOp,
                                   RDVFinalCleanupList &cl) {
   LDBG() << "Processing region branch op: "
          << OpWithFlags(regionBranchOp, OpPrintingFlags().skipRegions());
+
+  // Scenario 1. This is the only case where the entire `regionBranchOp`
```

Comment:
I just moved this further up, so that the early exit is closer to the beginning.